File: Codes from the AI-assistants/Old Experiments/[Experimental] regex-redux/Basic Prompt - TBR Mode/RegexRedux-deepseek-basic-tbr.js
const fs = require('fs');

const input = fs.readFileSync(0, 'utf-8'); // Read entire stdin
const ilen = input.length;

// Remove FASTA header and newlines
let cleaned = input.replace(/>.*\n|\n/g, '');
const clen = cleaned.length;

// Patterns for counting
const patterns = [
    "agggtaaa|tttaccct",
    "[cgt]gggtaaa|tttaccc[acg]",
    "a[act]ggtaaa|tttacc[agt]t",
    "ag[act]gtaaa|tttac[agt]ct",
    "agg[act]taaa|ttta[agt]cct",
    "aggg[acg]aaa|ttt[cgt]ccct",
    "agggt[cgt]aa|tt[acg]accct",
    "agggta[cgt]a|t[acg]taccct",
    "agggtaa[cgt]|[acg]ttaccct"
];

// Count occurrences for each pattern
const counts = [];
patterns.forEach(p => {
    const re = new RegExp(p, 'gi');
    const matches = cleaned.match(re);
    counts.push(matches ? matches.length : 0);
    console.log(`${p} ${counts[counts.length - 1]}`);
});

console.log(''); // Blank line
console.log(ilen);
console.log(clen);

// IUB replacements
const iubMap = {
    'B': '(c|g|t)',
    'D': '(a|g|t)',
    'H': '(a|c|t)',
    'K': '(g|t)',
    'M': '(a|c)',
    'N': '(a|c|g|t)',
    'R': '(a|g)',
    'S': '(c|g)',
    'V': '(a|c|g)',
    'W': '(a|t)',
    'Y': '(c|t)'
};
const iubRegex = new RegExp(`[${Object.keys(iubMap).join('')}]`, 'g');
const iubStr = cleaned.replace(iubRegex, match => iubMap[match]);
const iubLen = iubStr.length;

console.log(iubLen);

File: Codes from the AI-assistants/Old Experiments/[CLBG] regex-redux/Basic Prompt - TBR Mode/RegexRedux-deepseek-basic-tbr.py
import re
import sys

def main():
    data = sys.stdin.read()
    
    # Remove FASTA headers (lines starting with '>') and newlines
    seq = re.sub(r'>.*\n?', '', data)
    seq = re.sub(r'\n', '', seq)
    
    len_clean = len(seq)
    len1 = len_clean
    len2 = len_clean
    
    patterns = [
        r'agggtaaa|tttaccct',
        r'[cgt]gggtaaa|tttaccc[acg]',
        r'a[act]ggtaaa|tttacc[agt]t',
        r'ag[act]gtaaa|tttac[agt]ct',
        r'agg[act]taaa|ttta[agt]cct',
        r'aggg[acg]aaa|ttt[cgt]ccct',
        r'agggt[cgt]aa|tt[acg]accct',
        r'agggta[cgt]a|t[acg]taccct',
        r'agggtaa[cgt]|[acg]ttaccct'
    ]
    
    for pat in patterns:
        count = len(re.findall(pat, seq, flags=re.IGNORECASE))
        print(f"{pat} {count}")
    
    # Perform the replacements in order
    seq = re.sub(r'tHa[Nt]', '<4>', seq, flags=re.IGNORECASE)
    seq = re.sub(r'aND|caN|Ha[DS]|WaS', '<3>', seq, flags=re.IGNORECASE)
    seq = re.sub(r'a[NSt]|BY', '<2>', seq, flags=re.IGNORECASE)
    seq = re.sub(r'<[^>]*>', '|', seq)
    seq = re.sub(r'\|[^|][^|]*\|', '-', seq)
    
    len3 = len(seq)
    
    print()
    print(len1)
    print(len2)
    print(len3)

if __name__ == "__main__":
    main()